### PR TITLE
Rebased version of [WIP] Bundled cacert #4679

### PR DESCRIPTION
**Pull request by gvollant**
I added on my fork https://github.com/gvollant/curl a rebased version of #4679 
I invite @moparisthebest  on my curl fork. He made the real work

I just wanted a rebased version of this PR to test it against current curl master

Validation : 397e11bfa6175c525d9365e63162888a77008213 [397e11b]
Parents : 4506607b44
Auteur : moparisthebest <admin@moparisthebest.com>
Date : vendredi 3 avril 2020 00:32:42
Auteur : moparisthebest
Date de validation : samedi 4 avril 2020 00:31:23
Add CURLOPT_CAINFO_PEM

### CODE REVIEW

**Comment by gvollant** [comment]
this PR is probably exclusive with #5664

**Comment by nadrolinux** [comment]
Hello,
Any progress in this area? This is really nice feature.

**Comment by gvollant** [comment]
> 
> 
> Hello,
> Any progress in this area? This is really nice feature.

I just rebase it against current master

**Comment by bagder** [comment]
Note that this still causes test failures (test 1119) which is one of the primary reasons I've not checked it closer yet.

File: lib/vtls/openssl.c [hunk 1]
```diff
@@ -2987,6 +3043,15 @@ static CURLcode ossl_connect_step1(struct connectdata *conn, int sockindex)
...
   }
 #endif
+  if(ca_file_pem && load_cacert_from_memory(backend->ctx, ca_file_pem)) {
+    if(verifypeer) {
+      /* Fail if we insist on successfully verifying the server. */
+      failf(data, "error setting certificate from memory");
+      return CURLE_SSL_CACERT_BADFILE;
+    }
+    /* Continue with a warning if no certificate verification is required. */
+    infof(data, "error setting certificate file, continuing anyway\n");
```
Comment (JCMais) [inline lib/vtls/openssl.c:3053]: Should we really ignore the failure to load the cacert from memory when we are not verifying the peer certificate?

Would not it be better to fail anyway? Honest question, I don't have a specific opinion on this matter.

**Comment by bagder** [comment]
Closing due to inactivity.

**Comment by JCMais** [comment]
The test failure seems to be related to the SYMBOLS_IN_VERSIONS documentation page.

@gvollant @moparisthebest are you guys available to take a look into this? This feature seems very interesting.

If not, I should be able to take a look at it later next week.

**Comment by gvollant** [comment]
I rebased https://github.com/gvollant/curl/tree/gv_bundled_cacert but PR is closed...

**Comment by JCMais** [comment]
@gvollant It was closed due to inactivity. It was not reviewed before because the tests were failing: https://github.com/curl/curl/runs/1098854111

**Comment by gvollant** [comment]
The latest commit must fix test 1119 . How can we reopen ?

**Comment by JCMais** [comment]
cc @bagder

**Comment by bagder** [comment]
You can't. If you're set to work on it, submit a new PR!

**Comment by JCMais** [comment]
@gvollant let me know if have the time to open a new PR with the fixes. I'm eager for this feature, if no one is going to keep working on it I should be able to take ownership of the remaining changes needed.

**Comment by gvollant** [comment]
@JCMais see #6109
I send you invitation on my fork

**Pull request closed without merging**